oneview_redfish_toolkit/blueprints/storage.py

```diff
@@ -35,66 +35,84 @@
+        Internal Server Error.
+
+        Returns:
+            JSON: Redfish json with Storage
+            When server profile is not found calls abort(404)
+
+        Exceptions:
+            Logs the exception and call abort(500)
+
+    """
+
+    drive_id_int = None
+    logical_jbod = None
     try:
-        server_hardware = g.oneview_client.server_hardware. \
-            get(uuid)
-
-        sht_uri = server_hardware['serverHardwareTypeUri']
-        server_hardware_type = \
-            g.oneview_client.server_hardware_types.get(sht_uri)
-
-        st = Storage(uuid, server_hardware_type)
-
-        json_str = st.serialize()
-
-        return Response(
-            response=json_str,
-            status=status.HTTP_200_OK,
-            mimetype="application/json")
-    except HPOneViewException as e:
-        if e.oneview_response['errorCode'] == "RESOURCE_NOT_FOUND":
-            if e.msg.find("server-hardware-types") >= 0:
-                logging.warning(
-                    'Server hardware type ID {} not found'.
-                    format(server_hardware['serverHardwareTypeUri']))
-                abort(
-                    status.HTTP_404_NOT_FOUND,
-                    "Server hardware types not found")
-            else:
-                logging.warning(
-                    'Server hardware UUID {} not found'.
-                    format(uuid))
-                abort(
-                    status.HTTP_404_NOT_FOUND,
-                    "Server hardware not found")
-        elif e.msg.find("server-hardware-types") >= 0:
-            logging.exception(
-                'OneView Exception while looking for server hardware type'
-                ' {}'.format(e)
-            )
-            abort(status.HTTP_500_INTERNAL_SERVER_ERROR)
-        elif e.msg.find("server-hardware") >= 0:
-            logging.exception(
-                'OneView Exception while looking for '
-                'server hardware: {}'.format(e)
-            )
-            abort(status.HTTP_500_INTERNAL_SERVER_ERROR)
-        else:
-            logging.exception('Unexpected OneView Exception: {}'.format(e))
-            abort(status.HTTP_500_INTERNAL_SERVER_ERROR)
-    except Exception as e:
-        # In case of error print exception and abort
-        logging.exception('Unexpected error: {}'.format(e))
-        return abort(status.HTTP_500_INTERNAL_SERVER_ERROR)
+        drive_id_int = int(drive_id)
+    except ValueError:
+        abort(status.HTTP_400_BAD_REQUEST, "Drive id should be a integer")
+
+    server_profile = g.oneview_client.server_profiles.get(profile_id)
+    sas_logical_jbods = _find_sas_logical_jbods_by(server_profile)
+
+    count_drives = 0
+    for log_jbod in sas_logical_jbods:
+        next_count = count_drives + int(log_jbod["numPhysicalDrives"])
+
+        if drive_id_int in range(count_drives + 1, next_count + 1):
+            logical_jbod = log_jbod
+            break
+
+        count_drives = next_count
```

Comment:
About this logic, the sas_logical_jbods retrieved from OneView, are sequentially ordered by the "numPhysicalDrives"? Since this code walk through windows of intervals for the drivers, in order to get the logical_jbod that represents the drive_id_int, I think we need that the sas_logical_jbods are ordered by the "numPhysicalDrivers".

Reply:
I am assuming that sas_logical_jbods are ordered, but I will sort by ID to have sure.